test/test_ranked_queries.cpp

```diff
@@ -136,6 +136,16 @@ TEST_CASE_METHOD(pisa::test::index_initialization, "ranked_or_taat_lazy")
     test_against_or(query_fun);
 }
 
+TEST_CASE_METHOD(pisa::test::index_initialization, "range_query")
+{
+    auto query_fun = [&](pisa::term_id_vec terms){
+        pisa::range_query<pisa::wand_query> wand_q(10, index.num_docs(), 128);
```

Comment:
Can you add a test for each of the algorithms we have?
So one for `block_max_wand`, another one  for `maxscore` etc.

Reply:
We could do something smart in here (I'm writing off the top of my head, please consult documentation if decide to do it this way):
```cpp
TEST_CASE_METHOD(pisa::test::index_initialization, "range_query")
{
    // normally, we wouldn't erase for efficiency reasons, but here only testing correctness
    // BTW. in the coming versions of Catch2, there will be a template generator, so no erasure will be necessary
    using fn_type = std::function<uint64_t(decltype(make_block_max_scored_cursors(index, wdata, pisa::term_id_vec{})))>;
    auto query_type = GENERATE(fn_type, pisa::wand_query, pisa::maxscore_query, etc.);
    CAPTURE(query_type);
    auto query_fun = [&](pisa::term_id_vec terms){
        pisa::range_query<fn_type> q(10, index.num_docs(), 128);
        q(make_block_max_scored_cursors(index, wdata, terms));
        return q;
    };
    test_against_or(query_fun);
}
```

But if you're having problems with this, just copy/paste and modify.

Reply:
Yes, this is much nicer. Let's do it this way pls.